core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -196,14 +196,8 @@ public synchronized void connect() throws IOException, ConnectionFailedException
   public synchronized void disconnect() {
     if (mConnected) {
       LOG.debug("Disconnecting from the {} {} {}", getServiceName(), mMode, mAddress);
-      mConnected = false;
-    }
-    try {
       beforeDisconnect();
-      if (mProtocol != null) {
-        mProtocol.getTransport().close();
-      }
-    } finally {
+      mProtocol.getTransport().close();
```

Comment:
I read the source code, it seems  when `mConnected == true`, this variable will always be not null. discard my question.